app/controllers/ops_controller/ops_rbac.rb

```diff
@@ -670,6 +670,7 @@ def rbac_edit_save_or_add(what, rbac_suffix = what)
     when :group then
       record = @edit[:group_id] ? MiqGroup.find_by(:id => @edit[:group_id]) : MiqGroup.new
       validated = rbac_group_validate?
+      rbac_group_set_record_description_role(record) # Set new Description, Role and Tenant for a new Group
```

Comment:
first validate, then set extra fields?

Shouldn't this be the other way around?

Reply:
This was like that, originally. And yes, it looks weird but the logic behind this is that in `rbac_group_validate?`, it is just checked that user filled in required fields like _Description_ and _Role_. `@edit[:new]` is checked there, not `record`. And if everything is ok, then `record` is set according to what user filled in and then saved.

My change here is that I am putting the new info which is required (Description, Role, Tenant) to `record` even if `rbac_group_validate?` fails. It fails if any of required fields is missing, not if all of them. And if it fails, `rbac_group_set_record_vars` wouldn't be executed so `record` would miss ALL the info user filled in properly. So then later [here](https://github.com/ManageIQ/manageiq-ui-classic/pull/6169/files#diff-8078c760e06b820719715450b2a112e0L680), `record.description` would be `nil`, instead of the name user filled in. This leads to displaying error message about missing Description which does not make sense because user filled in Description (and forgot to choose a role, for example).

Reply:
Aah, perfect, thanks! :)